code/obj/item/gun/energy.dm

```diff
@@ -1027,12 +1027,42 @@
 	icon_state = "hunter"
 	item_state = "hunter"
 	base_item_state = "hunter"
+	var/hunter_key = "" // The owner of this rifle.
 
 	New()
 		..()
 		if(istype(src.loc, /mob/living))
 			var/mob/M = src.loc
 			src.AddComponent(/datum/component/self_destruct, M)
+			src.hunter_key = M.mind.key
+			START_TRACKING_CAT(TR_CAT_HUNTER_GEAR)
+			flick("[src.base_item_state]-tele", src)
+
+	disposing()
+		. = ..()
+		STOP_TRACKING_CAT(TR_CAT_HUNTER_GEAR)
+
+	proc/send_to_target_mob(var/mob/living/M)
```

Comment:
The copypasted proc is not very neat. Ideally I think this all would be a component. But if you don't feel like making it a component probably make the proc a global one instead.